modules/topology_hiding/topo_hiding_logic.c

```diff
@@ -1596,8 +1601,12 @@ static char* build_encoded_contact_suffix(struct sip_msg* msg, str *routes, int
 		ct_len = (short)contact.len;
 	}
 
+	flags_len = (short)sprintf(flags_char, "%d", flags);
```

Comment:
The practice is to use [int2str](https://github.com/OpenSIPS/opensips/blob/master/ut.h#L280) function to convert ints to string.